lib/awesome_spawn/command_result_error.rb

```diff
@@ -1,9 +1,13 @@
 module AwesomeSpawn
   class CommandResultError < StandardError
+    include AwesomeSpawn::Logging
+
     # @return [CommandResult] The command that caused the error
     attr_reader :result
 
     def initialize(message, result)
+      logger.error("AwesomeSpawn: #{message}")
+      logger.error("AwesomeSpawn: #{result.error}")
```

Comment:
Sorry I didn't mean callers that far out, I meant within AwesomeSpawn, which is just one place.